From ea1853ca01c01489a81e9662707bc257504bbbce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jari=20H=C3=A4m=C3=A4l=C3=A4inen?= <nuumiofi@gmail.com>
Date: Mon, 26 Sep 2022 15:47:49 +0300
Subject: [PATCH] panfrost: fix emulationstation and RetroArch menu fps drop

git bisect showed that commit 673f812835be26500958c79cd6f10bd24f99a385
causes fps drop in Batocera emulationstation and in RetroArch menu.

Fixed by commenting out the line added in that commit.

Tested with rk3399/rockpro64 target.
---
 src/gallium/drivers/panfrost/pan_resource.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/gallium/drivers/panfrost/pan_resource.c b/src/gallium/drivers/panfrost/pan_resource.c
index b673af74370..9c575f5a287 100644
--- a/src/gallium/drivers/panfrost/pan_resource.c
+++ b/src/gallium/drivers/panfrost/pan_resource.c
@@ -1061,7 +1061,8 @@ panfrost_ptr_map(struct pipe_context *pctx,
                                  * accessing this resource, flush them but do
                                  * not wait for them.
                                  */
-                                panfrost_flush_batches_accessing_rsrc(ctx, rsrc, "Resource shadowing");
+                                /* batocera: fix ES and RA menu FPS drop by commenting out next line */
+                                /* panfrost_flush_batches_accessing_rsrc(ctx, rsrc, "Resource shadowing"); */
 
 	                        if (!copy_resource &&
                                     drm_is_afbc(rsrc->image.layout.modifier))
-- 
2.37.3

